jobs/samba-integration.yml

```diff
@@ -0,0 +1,68 @@
+- job:
+    name: gluster_samba-integration
+    node: gluster
+    description: Run integrations for clustered samba on top of gluster.
+        They are run on latest bits from Samba and Gluster.  The test is run
+        periodically and on PRs to the samba-integration repository.
+    project-type: freestyle
+    concurrent: true
+
+    parameters:
+    - string:
+        default: '1'
+        description: Number of nodes for this test
+        name: NODE_COUNT
+    - string:
+        default: ''
+        description: ghprbPullId will be set when the build is triggered
+          through a GitHub pull-request, leave empty for running a test against
+          the master branch
+        name: ghprbPullId
+
+    scm:
+    - git:
+        url: https://github.com/gluster/samba-integration
+        branches:
+        - origin/centos-ci
```

Comment:
Changes on the master branch (that contain the actual tests) should be tested as well when a PR comes in.
In case the master branch is not ready yet, tests will fail, but that is fine too. At one point they should start succeeding.

Reply:
Yeah, I was not 100% clear what is the relationship between / meaning of  `scm->git` and `properties->github` ...

Reply:
I thought that the pull-request trigger would act on the `properties->github` instead of on the `scm->git`...

Reply:
I was guided by the heketi example:

```
...
    scm:
    - git:
        url: https://github.com/heketi/ci-tests
        branches:
        - origin/centos-ci

    properties:
    - github:
        url: https://github.com/heketi/heketi
    - build-discarder:
        days-to-keep: 7
        artifact-days-to-keep: 7

    triggers:
    - github-pull-request:
```

In this case, the integration script (corresponding to this one) is in the `heketi/ci-tests` repo (`scm->git`), while the tests are in the `heketi/heketi` repo (that's `properties->github`), and that is what is used to trigger runs from PRs.

So I was assuming that the way I set this up, it would trigger from the master branch of `gluster/samba-integration`... but the properties doesn't allow to specify the branch under github, right?

Reply:
So, isn't this going to trigger tests on PRs to all branches of samba-integration?

Reply:
I am not sure what the `branches` option does, it possibly can just be removed ([docs are unclear about it](https://jenkins-job-builder.readthedocs.io/en/latest/scm.html#scm.git))